cmake-modules/create_map_file.cmake

```diff
@@ -0,0 +1,16 @@
+# Instructs linker to generate map files and optimize build for minimal size
+# Requires CMake version >= 3.13 to use add_link_options
+
+function(create_map_file MAP_FILE_NAME)
+    if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.13")
```

Comment:
This might solve the letter of the issue, but not the spirit :)

That said, it is fine to defer the actual fix to keep this working even on 3.10 to later.

Reply:
Yes, this might come back to bite us later. I've added a message and filed an issue to track fixing this if we start building with CMake 3.10 all over and wonder where the map files went: https://github.com/Azure/azure-sdk-for-c/issues/1220